src/stirling/source_connectors/socket_tracer/bcc_bpf/protocol_inference.h

```diff
@@ -487,6 +487,66 @@ static __inline bool is_redis_message(const char* buf, size_t count) {
   return true;
 }
 
+static __inline enum message_type_t infer_mux_message(const char* buf, size_t count) {
+  static const int8_t kTreq = 1;
```

Comment:
I attempted to enhance the mux protocol inference by reducing the number of types (R/Tinit, R/Tdispatch, Rerr(Old)) and by parsing 'special' strings that are present in those frame types. Ideally we would support all mux types, but there are some that aren't as high value.

If I remember correctly, the protocol inference runs on a connection until it's classified and is only provided the data from the syscall, correct? If that's the case I was hoping that having stronger verification on Rerr, Tinit and Rinit would help reduce false positives since it's earlier in the connection lifecycle. Let me know what you think of my latest update.

I am definitely interested in trying the protocol inference against the network traffic dataset. I shared the mux pcap file that I generated myself with you in slack. Unfortunately I don't know if I'll be able to provide real application traffic, but let me know how I can help with this testing.